apps/src/templates/GDPRDialog.jsx

```diff
@@ -0,0 +1,82 @@
+import React, {Component, PropTypes} from 'react';
+import Button from './Button';
+import i18n from "@cdo/locale";
+import BaseDialog from './BaseDialog';
+import DialogFooter from "./teacherDashboard/DialogFooter";
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
+
+const styles = {
+  dialog: {
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  },
+  instructions: {
+    marginTop: 20
+  }
+};
+
+export default class GDPRDialog extends Component {
+  static propTypes = {
+    isDialogOpen: PropTypes.bool.isRequired,
+    currentUserId: PropTypes.number.isRequired,
+    requestIp: PropTypes.string.isRequired,
+    // If we're coming from pegasus, studioUrlPrefix is passed in as a prop and used to construct the logout url.
+    studioUrlPrefix: PropTypes.string
+  };
+
+  state = {
+    isDialogOpen: this.props.isDialogOpen,
+  };
+
+  handleYesClick = () => {
+    this.setState({isDialogOpen: false});
+    $.post(
+      `/api/v1/users/${this.props.currentUserId}/accept_data_transfer_agreement`, {
+        data_transfer_agreement_request_ip: this.props.requestIp,
+        data_transfer_agreement_source: "DIALOG",
```

Comment:
Maybe something more specific to this dialog?  Say "ACCEPT_DATA_TRANSFER_DIALOG"?